src/services/v0/v0blocks/BlocksService.ts

```diff
@@ -395,26 +399,34 @@ export class BlocksService extends AbstractService {
 				const argument = callArgs.get(paramName);
 
 				if (Array.isArray(argument)) {
-					newArgs[paramName] = this.parseArrayGenericCalls(argument);
+					newArgs[paramName] = this.parseArrayGenericCalls(
+						argument,
+						block
+					);
 				} else if (argument instanceof GenericCall) {
-					newArgs[paramName] = this.parseGenericCall(argument);
+					newArgs[paramName] = this.parseGenericCall(argument, block);
 				} else if (
 					paramName === 'call' &&
 					argument?.toRawType() === 'Bytes'
 				) {
 					// multiSig.asMulti.args.call is an OpaqueCall (Vec<u8>) that we
 					// serialize to a polkadot-js Call and parse so it is not a hex blob.
-					const call = this.api.createType('Call', argument.toHex());
-					newArgs[paramName] = this.parseGenericCall(call);
+					const call = block.registry.createType(
+						'Call',
+						argument.toHex()
+					);
+					newArgs[paramName] = this.parseGenericCall(call, block);
 				} else {
 					newArgs[paramName] = argument;
 				}
 			}
 		}
 
 		return {
-			method: `${sectionName}.${methodName}`,
-			callIndex,
```

Comment:
Why remove the callIndex here? It’s a private method so I guess it can change at any time, but curious why?

Reply:
Thats a mistake! Good catch!